src/value.rs

```diff
@@ -90,6 +102,65 @@ impl AllocatedValue {
     }
 }
 
+impl SignedInteger {
+    pub fn zero() -> Self {
+        SignedInteger::Positive(0)
+    }
+
+    pub fn to_u64(&self) -> Option<u64> {
+        match self {
+            SignedInteger::Positive(x) => Some(*x),
+            SignedInteger::Negative(_) => None,
+        }
+    }
+
+    fn to_i128(&self) -> i128 {
+        match self {
+            SignedInteger::Positive(x) => (*x).into(),
+            SignedInteger::Negative(x) => -(*x as i128),
+        }
+    }
+
+    fn from_i128(x: i128) -> Self {
+        if i128::is_negative(x) {
```

Comment:
I’m curious if we can do this choice in const time.

Reply:
Is the function not constant time right now because of the `if`/`else` block?